packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -409,100 +437,158 @@ function useMediaUploadQueue() {
-            ITEM_STATUS.TRIMMING,
-          ].includes(item.state) && item.originalResourceId === resourceId
+          item.state !== ITEM_STATUS.CANCELLED &&
+          item.originalResourceId === resourceId
       );
+
     /**
-     * Is the current resource being processed.
+     * Determine whether the current resource is being processed.
      *
-     * @param {number} resourceId Resource id.
-     * @return {boolean} if resource with id is found.
+     * This is the case when uploading a new media item that first
+     * needs to be transcoded/muted/trimmed.
+     *
+     * As long as that resource's upload hasn't been cancelled,
+     * it is considered to be processing.
+     *
+     * @param {number} resourceId Resource ID.
+     * @return {boolean} Whether the resource is processing.
      */
     const isCurrentResourceProcessing = (resourceId) =>
       state.queue.some(
         (item) =>
-          [
-            ITEM_STATUS.PENDING,
-            ITEM_STATUS.UPLOADING,
-            ITEM_STATUS.TRANSCODING,
-            ITEM_STATUS.MUTING,
-            ITEM_STATUS.TRIMMING,
-          ].includes(item.state) && item.resource.id === resourceId
+          item.state !== ITEM_STATUS.CANCELLED &&
+          (item.resource.id === resourceId ||
+            item.previousResourceId === resourceId)
       );
+
     /**
-     * Is the current resource uploading.
+     * Determine whether the current resource is being uploaded.
      *
-     * @param {number} resourceId Resource id.
-     * @return {boolean} if resource with id is found.
+     * This is the case when uploading a new media item after any initial
+     * transcoding/muting/trimming has already happened.
+     *
+     * Checks for both `resource.id` as well as `previousResourceId`,
+     * since after upload to the backend, the resource's temporary uuid
+     * will be replaced by the permanent ID from the backend.
+     *
+     * @param {number} resourceId Resource ID.
+     * @return {boolean} Whether the resource is uploading.
      */
     const isCurrentResourceUploading = (resourceId) =>
       state.queue.some(
         (item) =>
-          item.state === ITEM_STATUS.UPLOADING &&
-          item.resource.id === resourceId
+          [
+            ITEM_STATUS.PENDING,
+            ITEM_STATUS.UPLOADING,
+            ITEM_STATUS.UPLOADED,
+          ].includes(item.state) &&
+          (item.resource.id === resourceId ||
+            item.previousResourceId === resourceId)
       );
+
     /**
-     * Is the current resource transcoding.
+     * Determine whether the current resource is being transcoded.
      *
-     * @param {number} resourceId Resource id.
-     * @return {boolean} if resource with id is found.
+     * This is the case when uploading a new media item that first
+     * needs to be transcoded.
+     * This is also the case when optimizing an existing video in the story,
+     * which will cause a new resource to be uploaded (the "current" one).
```

Comment:
Below, for `isResourceTranscoding`, the comment is the following: 

> When optimizing an existing video in the story, which will cause a new resource to be uploaded, this returns the state of this new resource.

So this means that `isResourceTranscoding` is for checking the newly added resource, right?

Here, it says this:

> This is also the case when optimizing an existing video in the story, which will cause a new resource to be uploaded (the "current" one)

I read out from here that the new resource to be uploaded is the "current" one (since it's mentioned in the parenthesis). Should `(the "current" one)` be after the `"...when optimizing an existing video"` instead, if `isCurrentResourceTranscoding` is for the origin video?

Reply:
Sounds like the naming is still confusing despite (or because of) the comments.

I'll revisit :)

Reply:
Renamed functions & improved docs